app/assets/javascripts/validation.js

```diff
@@ -180,10 +189,15 @@ function validatePassword(confirmPasswordElement, obj) {
 }
 
 function validateConfirmPassword(passwordElement, obj) {
-  var confirmPassword = this.value;
+  var confirmPassword = this.value.trim();
   var password = passwordElement.value;
 
-  if (confirmPassword !== password) {
+  if (confirmPassword.length === 0) {
+    obj.updateUI(this, false, "This field can't be empty");
+    return;
+  }
+
+  if (confirmPassword !== password || !isPasswordValid(password)) {
     obj.updateUI(this, false, "Passwords must be equal");
```

Comment:
I think this should say "Passwords should be the same.". I totally get why you did that though. You see the `confirmPassword !== password` and you start thinking like the computer :smile:

Reply:
@Uzay-G I think that should is not meant to be placed here.
If the user want to complete the form successfully, passwords must(have to) be equal.
Should is not that enforcing. 

@SidharthBansal your opinion on this? Thank you both.

Reply:
Oh no that's not I was focusing on. It should be: `Passwords should be the same.`

Reply:
I just meant it should "the same" instead of "equal". I totally agree that we should use `must`

Reply:
In that case, I apologise. Let me quickly update this.